features/netsocket/mbed_lib.json

```diff
@@ -40,7 +40,7 @@
         },
         "default-mesh-type": {
             "help": "Configuration type for MeshInterface::get_default_instance(). [LOWPAN/THREAD/WISUN]",
-            "value": "THREAD"
+            "value": null
```

Comment:
one question before integrating this: isn't this breaking change for apps - as the config value changes for default mesh type ?

Reply:
As discussed, these changes are potentially breaking, adding proper labels

Reply:
Questionable..
Is it a breaking change to return no default, instead of non-configured non-working default?

Or @mikter did the non-configured Thread work? My assumption is that it is a demo, with no real use cases.. should we optimise for testing and demo, or for real use cases.

Reply:
I don't think anyone should have Thread configured without any configuration in application 

It would however work as a non-commissioned Thread device, but that should still be application configuration not a default. 

I can remove this change if you like from this commit as it is not really related. although the Breaking change is not only because of this configuration, but it is because we are changing status code on public API.